userguide/tutorials/invoice_examples.adoc

```diff
@@ -0,0 +1,169 @@
+
+So, `Invoice Balance` can be expressed as follows:
+
+[source,java]
+INVOICE BALANCE = CHARGED AMOUNT - AMOUNT PAID - AMOUNT REFUNDED - SUM(CBA_ADJ)
+
+[NOTE]
+*Note:* In the case of an invoice credit, the sum of the `CREDIT_ADJ` items is also added. This is done to keep the invoice balance `0` as explained in the invoice credit scenario below.
+
+
+The invoice balance is *always* 0 if:
+
+* The invoice is in a `DRAFT` or `VOID` status
+* The invoice is tagged as `WRITTEN_OFF` specifying that its balance should be `0`
+* The invoice is a migration invoice (imported from a different billing system into Kill Bill)
+* The invoice corresponds to a child account and has a zero parent invoice balance
+
+Let us now understand how the `charged amount` and `invoice balance` are computed in various scenarios.
+
+== Creating a recurring subscription
+
+When a recurring subscription is billed, the system generates an invoice corresponding to it. For example, consider the https://github.com/killbill/killbill-docs/blob/4671dcd9da1cf021e85629ab67e3ffb6fb553bb1/catalogs/monthly-with-trial.xml[monthly-with-trial] catalog. It has a `standard-monthly` plan with a `TRIAL` phase of `10` days followed by an `EVERGREEN` phase with a monthly recurring charge of `$24.95`. Let us understand what happens in this scenario. 
+
+. As soon as a subscription is created, it is in `TRIAL` phase. An invoice is generated with one invoice item of type `FIXED` and amount `0`. 
+
+So, `CHARGED AMOUNT` and `BALANCE` are calculated as follows:
+
+[source,java]
+----
+CHARGED AMOUNT = FIXED + RECURRING + EXTERNAL_CHARGE + USAGE + TAX + ITEM_ADJ + REPAIR_ADJ + CREDIT_ADJ
+               = 0 + 0 + 0 + 0 + 0 + 0 + 0
+               = 0
+               
+INVOICE BALANCE = CHARGED AMOUNT - AMOUNT PAID - AMOUNT REFUNDED - SUM(CBA_ADJ)
+                = 0 - 0 - 0 - 0
+                = 0                
+----
+
+[start=2]
+. Once the `TRIAL` phase ends, the subscription moves to the `EVERGREEN` phase. An invoice is generated with one invoice item of type `RECURRING` and amount `24.95`.
+
+So, `CHARGED AMOUNT` and `BALANCE` are calculated as follows:
+
+[source,java]
+----
+CHARGED AMOUNT = FIXED + RECURRING + EXTERNAL_CHARGE + USAGE + TAX + ITEM_ADJ + REPAIR_ADJ + CREDIT_ADJ
+               = 0 + 24.95 + 0 + 0 + 0 + 0 + 0 + 0
+               = 24.95
+               
+INVOICE BALANCE = CHARGED AMOUNT - AMOUNT PAID - AMOUNT REFUNDED - SUM(CBA_ADJ)  
+                = 24.95 - 24.95 - 0 - 0   // assuming the account has a valid default payment method, AMOUNT PAID=24.95
+                = 0
+----
+             
+== Adding an invoice credit
+
+Whenever a https://killbill.github.io/slate/#credit-create-credits[credit is added], the system generates an invoice credit. An invoice credit is nothing but an invoice with a `CREDIT_ADJ` and a `CBA_ADJ` item. When such a credit is added, it is adjusted against the amount/balance in other unpaid invoices (if any). 
```

Comment:
I don't know if we should talk about 'invoice credit' or 'account credit'. @pierre Any thoughts?

Reply:
Changed section title to `Add Credit` and added separate cases for `Account Credit` and `Invoice Credit`. Let me know how it looks